### ISSUE TITLE
[Bug]: Allowed brick types are out-of-date for versions

### ISSUE THREAD

**Issue by BlackbitDevs** [pimcore/pimcore#15251]
### Pimcore version

\>= 6

### Steps to reproduce

Similar to #2585 it is a problem that the allowed bricks get saved in a class property (and not in a getter method) because when objects get serialized, the property `$allowedTypes` will be hard-coded to the list of bricks which were available at the time when the object got saved. If later another brick gets allowed for the object brick container field and the version data gets loaded, it will not be possible to add the new brick.

Steps to reproduce:
1. Create class `Test` with object brick container field `bricks`
2. Create object brick `BrickA` and allow it for above field
3. Create object `/test` of above class, save draft version
4. Create new brick `BrickB` and allow it for above field
5. Run the following script:
```php
<?php
$object = \Pimcore\Model\DataObject\Concrete::getByPath('/test');
$version = $object->getLatestVersion();
$versionObject = $version->getData();

var_dump($versionObject->getBricks()->getAllowedBrickTypes());
```

This will return 
```
array(3) {
  [0]=>
  string(6) "BrickA"
}
```

### Expected / actual behavior

It should be possible to add newly created object bricks to restored version objects.

**Comment by aryaantony92** [comment]
@BlackbitDevs Thanks for reporting the issue. Could you please provide the version files also ?

**Comment by BlackbitDevs** [comment]
@aryaantony92 I have added a detail in above steps to reproduce the problem: Of course you have to allow `BrickB` to `Test.bricks` field.

Currently I do not have any version file ready. But can't you reproduce the problem with above steps? The version will automatically be created when you save the draft version (step 3).

**Comment by dvesh3** [comment]
Fixed by #15963
